rpmio/macro.c

```diff
@@ -1096,6 +1096,22 @@ static void doLua(MacroBuf mb,  rpmMacroEntry me, ARGV_t argv, size_t *parsed)
     }
 }
 
+/*
+ * Wrap a call to Lua string functions.
+ * Note extra parentheses to force only one result returned, multi-value
+ * returns such as from string.gsub() make no sense in this context.
+ */
+static void doString(MacroBuf mb, rpmMacroEntry me, ARGV_t argv, size_t *parsed)
+{
+    char *arg = argvJoin(argv+1, "', '");
+    char *s = rstrscat(NULL,
+		"return (string.", argv[0], "('", arg, "'))", NULL);
```

Comment:
Eew, that's horrible. No support for strings with `"` in them?

Reply:
How about using `"return (string.", argv[0], "(table.unpack(arg)))"`? You need to directly call rpmluaRunScript(), of course.

Reply:
Heh, quite frankly I didn't even consider quote issue. It's ugly sure, it's the first solution my mind spat out on a Monday morning :laughing:

I was just looking at that table.unpack() stuff but didn't realize it was doLua() getting in the way.

Reply:
And yeah, that works, thanks for the suggestion.